monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/HttpClientPool.java

```diff
@@ -52,13 +52,21 @@ protected HttpClientPool(
         // For the new IPs new Http clients will be created
         final long dnsLookupRefreshPeriodInSeconds = serverConfiguration.getDnsLookupRefreshPeriodInSeconds();
         this.scheduledFutureDnsRefresh = scheduledExecutorService.scheduleAtFixedRate(
-                () -> refreshTheList(dnsLookupWrapper, serverConfiguration, httpClientsCache, singleHttpClientProvider, scheduledExecutorService),
+                () -> refreshTheListWrappedInTryCatch(
+                        dnsLookupWrapper,
+                        serverConfiguration,
+                        httpClientsCache,
+                        singleHttpClientProvider,
+                        scheduledExecutorService
+                ),
                 dnsLookupRefreshPeriodInSeconds,
                 dnsLookupRefreshPeriodInSeconds,
                 TimeUnit.SECONDS
         );
 
-        refreshTheList(dnsLookupWrapper, serverConfiguration, httpClientsCache, singleHttpClientProvider, scheduledExecutorService);
+        // We invoke the same method here as in the scheduler. We don't want constructor to crush in case of a temporary issue.
+        // However for misconfiguration problems it will re-throw an exception and crush the constructor.
+        refreshTheListWrappedInTryCatch(dnsLookupWrapper, serverConfiguration, httpClientsCache, singleHttpClientProvider, scheduledExecutorService);
```

Comment:
I'm not sure this one should be wrapped in a try/catch since it could lead to `null` list?

Reply:
It doesn't seem to be an issue as we already have a protection there:
```
return Optional.ofNullable(httpClientsCache.get()).orElse(RoundRobinPool.EMPTY);
```